python/tank/util/version.py

```diff
@@ -150,28 +186,31 @@ def suppress_known_deprecation():
         yield ctx
 
 
-def version_parse(version_string):
+def normalize_version_format(version):
     """
-    Parse a version string into a Version object. We also support LooseVersion
-    for compatibility with older versions of Python.
-
-    :param str version_string: The version string to parse.
-
-    :rtype: packaging.version.Version, LooseVersion or str as fallback.
+    Normalize version strings by applying common format transformations.
+
+    This function exists because packaging.version.parse() follows PEP 440
+    and cannot handle non-standard version formats like "v1.2.3" or "6.3v6",
+    which are commonly found in various software tools and DCCs but don't
+    conform to the PEP 440 specification.
+
+    Transformations applied:
+    - Strip whitespace and convert to lowercase
+    - Remove leading 'v' prefix
+    - Extract version numbers from software names: "Software Name 21.0" -> "21.0"
+    - Convert dot-v format: "6.3v6" -> "6.3.6"
+    - Convert simple v format: "2019v0.1" -> "2019.0.1"
+    - Convert service pack formats: "2017.2sp1" -> "2017.2.post1", "2017.2.sp1" -> "2017.2.post1"
+
+    :param str version_string: Version string to normalize
+    :return str: Normalized version string compatible with PEP 440
     """
-    if "packaging" in sys.modules:
-        try:
-            return packaging.version.parse(version_string)
-        except packaging.version.InvalidVersion:
-            # Version cannot be parsed with packaging.version (SG-40480)
-            pass
 
-    if LooseVersion:
-        with suppress_known_deprecation():
-            return LooseVersion(version_string)
+    for compiled_pattern, replacement in _VERSION_PATTERNS:
+        version = compiled_pattern.sub(replacement, version)
```

Comment:
Do we want to chain the substitution or we want to apply only once? 

If so:
```suggestion
    for compiled_pattern, replacement in _VERSION_PATTERNS:
        return compiled_pattern.sub(replacement, version)
```

Reply:
Chaining is essential because we don't know which pattern will match a given version string. The input "2019v0" needs to reach the third pattern (simple v format) to be transformed into "2019.0" - a valid PEP 440 format. Without chaining, versions that don't match the first pattern remain unnormalized and break downstream parsing. The loop ensures every version gets processed by all applicable patterns until it reaches a valid format.